tests/components/switchbot/test_fan.py

```diff
@@ -0,0 +1,164 @@
+"""Test the switchbot fan."""
+
+from collections.abc import Callable
+from unittest.mock import AsyncMock, patch
+
+from homeassistant.components.fan import (
+    ATTR_OSCILLATING,
+    ATTR_PERCENTAGE,
+    ATTR_PRESET_MODE,
+    DOMAIN as FAN_DOMAIN,
+    SERVICE_OSCILLATE,
+    SERVICE_SET_PERCENTAGE,
+    SERVICE_SET_PRESET_MODE,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.core import HomeAssistant
+
+from . import CIRCULATOR_FAN_SERVICE_INFO, make_advertisement
+
+from tests.common import MockConfigEntry
+from tests.components.bluetooth import inject_bluetooth_service_info
+
+
+async def test_circulator_fan_controlling(
+    hass: HomeAssistant, mock_entry_factory: Callable[[str], MockConfigEntry]
+) -> None:
+    """Test setting up the circulator controlling."""
+    inject_bluetooth_service_info(hass, CIRCULATOR_FAN_SERVICE_INFO)
+
+    entry = mock_entry_factory(sensor_type="circulator_fan")
```

Comment:
I would recommend to use `pytest.mark.parametrize` for this test so you can make the test smaller and DRY